woocommerce/google-listings-and-ads
# Clean up after upgrading wp-scripts to 22.1.0 and webpack config to v5

### Changes proposed in this Pull Request:

It's stage 3 of #1333 and also includes some relevant enhancements.

- Replace `file-loader` with the built-in Asset Modules of webpack to handle the exporting of images.
- Remove the `injectPolyfill: true` config of DEWP to drop support for IE.
   - Ref: https://github.com/WordPress/gutenberg/pull/35436
- Generate source map of SCSS in development.
- Use `<rootDir>` for paths in jest config, and ignore the **tests** folder when generating the coverage report.
- Remove temporary peer packages that have been correctly listed in those packages that require them.
   - `@wordpress/core-data`
   - `md5`
   - `postcss`
- Remove two `@wordpress/scripts` related packages that can be specified by `import/core-modules` settings of eslint.
   - `stylelint`
   - `@wordpress/stylelint-config`
- Deduplicate the content of **package-lock.json** file via `npm dedupe`.
   - 💡 After the deduplication, the package-lock file will have some diff when back and forth running `npm install` and `npm dedupe`. So the revision be87fa922f66ec901fea526389ef567eb79f89f4 was committed with the result of `npm dedupe` and then `npm install`.

### Detailed test instructions:

1. Run `npm install` to check if there is no error.
1. Run `npm run test:js`. Check if the **tests** folder is no longer included in the coverage report.
1. Run `npm run lint:js`. The eslint should be able to run.
1. Run/restart `npm start`. Browse GLA pages to check if the images are still shown.
1. Inspect some DOM elements to see if their CSS classes are referenced to the relatively correct number of lines of SCSS source code rather than the number of lines after compilation.
   - 💡 Please note that the number of lines displayed on DevTool will be the actual number of lines in the source file plus 1. Because each scss file will have an extra line inserted with SCSS `at-rules` at the file head via [this webpack config](https://github.com/woocommerce/google-listings-and-ads/blob/be87fa922f66ec901fea526389ef567eb79f89f4/webpack.config.js#L180-L185).

### Changelog entry
